network/shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/BlockTransferMessage.java

```diff
@@ -40,7 +41,8 @@
 
   /** Preceding every serialized message is its type, which allows us to deserialize it. */
   public static enum Type {
-    OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4);
+    OPEN_BLOCKS(0), UPLOAD_BLOCK(1), REGISTER_EXECUTOR(2), STREAM_HANDLE(3), REGISTER_DRIVER(4),
+    HEARTBEAT(5);
```

Comment:
Thanks @bbossy for testing this. the `sbin/start-shuffle-service` wasn't removed because we didn't want to break people's startup scripts. The Mesos one was added later, with the driver connection, etc. Something to be removed in 2.0!